deploy/charts/bottlerocket-shadow/templates/custom-resource-definition.yaml

```diff
@@ -0,0 +1,187 @@
+apiVersion: apiextensions.k8s.io/v1
```

Comment:
Possibly I'll answer this myself as I keep reading, but in the pre-helm setup, this was autogenerated based on our Rust models. Do we manually keep the two in sync now, or is it still possible to generate this document?

Reply:
Just to flesh this feedback out, a lot of these values only make sense in the context of the Kubernetes YAML model and have no bearing on the `brupop` sources, seems totally fine to remove the Rust models.

What I'm mostly interested in is the auto-generated enums and validations based on our Rust models, which was pretty handy. It would be a lot easier to have a mismatch or mistake if we untie them. Reading more carefully, it seems like the shadow CRD is still autogenerated (probably for this reason!). Nice!

Reply:
> What I'm mostly interested in is the auto-generated enums and validations based on our Rust models, which was pretty handy. It would be a lot easier to have a mismatch or mistake if we untie them. Reading more carefully, it seems like the shadow CRD is still autogenerated (probably for this reason!). Nice!

Right - I couldn't 100% decouple the kubernetes resources from the rust code. However, I think I struck a balance between what our rust code needs in it (like the enums, etc.) and yaml kubernetes resources

Reply:
Yes - let's move these togeher